inference-engine/src/mkldnn_plugin/nodes/embedding_bag_sum.cpp

```diff
@@ -0,0 +1,369 @@
+                    L(to_end);
+                } else {
+                    uni_vaddps(v_dst, v_dst, v_src);
+                }
+                uni_vmovups(ptr[reg_dst], v_dst);
+                add(reg_src, stride);
+                add(reg_dst, stride);
+            }
+        }
+
+        /*size_t work_rest = _jcp.emb_dim % v_len;
+        if (work_rest > 0) {
+            const size_t stride = sizeof(float);
+            if (_jcp.with_weights) {
+                fld(ptr[reg_weights]);
+            }
+            for (size_t i = 0; i < work_rest; i++) {
+                //mov(reg_v, ptr[reg_src]);
+                fld(ptr[reg_src]);
+                if (_jcp.with_weights) {
+                    fmul(st0, st1);
+                }
+                fadd(ptr[reg_dst]);
+                fstp(ptr[reg_dst]);
+                add(reg_src, stride);
+                add(reg_dst, stride);
+            }
+        }*/
+
+        this->postamble();
+        _ker = (decltype(_ker)) this->getCode();
+    }
+
+protected:
+    using Vmm = typename conditional3<isa == avx512_common, Xbyak::Zmm, isa == avx2, Xbyak::Ymm, Xbyak::Xmm>::type;
+
+    Vmm v_weights = Vmm(13);
+    Vmm v_src = Vmm(14);
+    Vmm v_dst = Vmm(15);
+
+    Xbyak::Reg64 reg_v = r11;
+    Xbyak::Reg64 reg_w = r12;
+    Xbyak::Reg64 reg_ww = r13;
+};
+
+
+const std::set<size_t> MKLDNNEmbeddingBagSum::_supported_indexes_type_size = {sizeof(INT32), sizeof(INT64)};
+
+MKLDNNEmbeddingBagSum::MKLDNNEmbeddingBagSum(
+        const CNNLayer* layer,
+        size_t required_input_num,
+        size_t indices_idx,
+        size_t per_sample_weights_idx,
+        size_t default_index_idx,
+        const std::set<Precision>& supported_precisions) :
+            INDICES_IDX(indices_idx),
+            PER_SAMPLE_WEIGHTS_IDX(per_sample_weights_idx),
+            DEFAULT_INDEX_IDX(default_index_idx) {
+    try {
+        std::string log_prefix = "Layer EmbeddingBagSum with name '";
+        if (layer->insData.size() < required_input_num || layer->outData.size() != 1)
+            THROW_IE_EXCEPTION << log_prefix << layer->name << "' has incorrect number of input or output edges!";
+        _l_name = layer->name;
+
+        auto inData = layer->insData[0].lock();
+        auto indicesData = layer->insData[INDICES_IDX].lock();
+        if (inData == nullptr || indicesData == nullptr)
+            THROW_IE_EXCEPTION << log_prefix << _l_name << "' layer has nullable input data.";
+
+        const auto precision = inData->getTensorDesc().getPrecision();
+        if (!supported_precisions.empty()) {
+            if (supported_precisions.find(precision) == supported_precisions.end())
+                THROW_IE_EXCEPTION << log_prefix << _l_name << "' layer has unsupported precision: " << precision.name();
+        } else {
+            static const std::set<Precision> default_supported_precisions =
+                {Precision::FP32, Precision::I8, Precision::U8, Precision::I32};
```

Comment:
We need to force FP32 in case input or output have BF16 precision.

Reply:
Fixed